android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt

```diff
@@ -0,0 +1,184 @@
+package com.mapbox.navigation.examples.androidauto
+
+import androidx.car.app.Session
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.androidauto.screenmanager.MapboxScreen
+import com.mapbox.androidauto.screenmanager.MapboxScreenEvent
+import com.mapbox.androidauto.screenmanager.MapboxScreenManager
+import com.mapbox.maps.logI
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.dropin.NavigationView
+import com.mapbox.navigation.dropin.navigationview.NavigationViewListener
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+/**
+ * This is a temporarily solution for syncing two new libraries, Drop-in-ui and the Mapbox AA.
+ *
+ * The libraries are defining public apis so that there can be options to determine the experience
+ * while both the car and phone are displayed.
+ */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CarAppSyncComponent private constructor() : MapboxNavigationObserver {
```

Comment:
Found the following scenario when it doesn't work:
1. Launch Android Auto and go to route preview. 
2. Launch the app on the phone. 
3. Android Auto unexpectedly transitions to Free Drive state.

Reply:
👍 nice find! will follow up on these fixes because it will be nice to also update the examples repo

Reply:
Ok tested this, making note of one extra step 

This works
1. Force kill app on the phone
2. ..

This reproduces the error
1. Launch app
2. Close the app
3. ...